File: 2019/Thang02_2018/24_02_2019/cards.cpp
#include <iostream>
#include <stdio.h>

using namespace std;
///////

const
    int maxn=1e5+100;
int n, k, m;
int xx[maxn], yy[maxn], nnum[maxn];
///////

void enter()
{
    scanf("%d%d%d", &n, &k, &m);
    for (int i=1; i<=m; i++) scanf("%d%d%d", &xx[i], &nnum[i], &yy[i]);
}
///////

int dequy(int i, int pos)
{
    if (i==0) return pos;
    int x=xx[i], y=yy[i], num=nnum[i];
    if (y<=pos && pos<=y+num-1) pos=x+pos-y; else
    if (pos<y)
    {
        if (pos>=x) pos+=num; else pos=pos;
    } else
    if (pos>y+num-1)
    {
        pos-=num;
        if (pos>=x) pos+=num; else pos=pos;
    }
    return dequy(i-1, pos);
}

void process()
{
    for (int i=1; i<=k; i++) cout << dequy(m, i) << ' ';
}
///////

int main()
{
    freopen("cards.inp", "r", stdin);
    freopen("cards.out", "w", stdout);
    enter();
    process();
}

File: 2018/Thang10-2018/12-10-2018/cards.pas
Const   ginp='cards2.inp';
        gout='cards.out';
        maxn=100010;

Type    save= record x,y,num:longint; end;

Var     n,m,k:longint;
        a:array[0..maxn] of save;

Function Try(i,pos:longint):longint;
Begin
        if i=0 then exit(pos);
        with a[i] do
        if (x<=pos) and (pos<=x+num-1) then pos:=pos+y-x else
        if (pos<x) and (pos>=y) then pos:=pos+num else
        if (pos>x+num-1) then
        begin
                pos:=pos-num;
                if pos>=y then pos:=pos+num;
        end;
        try:=try(i-1,pos);
End;

Procedure Enter;
Var     i:longint;
Begin
        readln(n,k,m);
        for i:=1 to m do readln(a[i].y,a[i].num,a[i].x);
End;

Procedure Process;
Var     i:longint;
Begin
        for i:=1 to k do
                write(try(m,i),' ');
End;

Begin
        Assign(input,ginp); Assign(output,gout);
        Reset(input); Rewrite(output);
        Enter;
        Process;
        Close(input); Close(output);
End.
